docs/dev/object_storage.md

```diff
@@ -112,3 +112,149 @@ Currently only snapshot backup is possible, so first one needs to take [snapshot
+which help gain local access to the data in case there is a need for manual intervention.
+
+Most of the time it won't be necessary to touch the data on S3 directly, there are transparent REST APIs and Scylla Manager  
+commands for backup and restore and Scylla can operate normally with S3 storage configured in the  
+`CREATE KEYSPACE` cql documented at [ScyllaDB CQL Extensions | ScyllaDB Docs](https://opensource.docs.scylladb.com/stable/cql/cql-extensions.html#keyspace-storage-options).  
+
+However, if for some reason the SSTables become corrupted and need an offline scrub before re-uploading  
+or if a bug investigation leads to the need to analyze the backup data, follow the information below to access  
+that data.  
+
+Issue tracking the document [here](https://github.com/scylladb/scylladb/issues/22438).
+
+## Object Storage Layout
+
+There are currently three mechanisms in Scylla which write data to S3:
+
+1. Scylla Manager backup
+
+When performing a backup with `sctool`, a `backup` prefix is created within the bucket passed as argument and  
+under that prefix, Scylla Manager stores all the backup data of all the backup tasks organized by cluster name,  
+datacenter, keyspace, etc.
+
+Follow [Specification | ScyllaDB Docs](https://manager.docs.scylladb.com/branch-3.3/backup/specification) in the Scylla Manager documentation for the exact layout  
+under the `backup` prefix.
+
+2. `/storage_service/backup` REST API
+
+When using the `/storage_service/backup` REST API, the data is stored under the prefix passed as argument to the API.  
+The structure under this prefix is identical to what you’d find in the typical Scylla snapshot.  
+There is a manifest file which contains the list of Data files for each SSTable, the schema file and all the SSTables  
+components stored flat under the prefix.
+```perl
+scylla-bucket/prefix/
+│
+├── manifest.json
+├── schema.cql
+|
+├── me-3gqe_1lnj_4sbpc2ezoscu9hhtor-big-Data.db
+├── me-3gqe_1lnj_4sbpc2ezoscu9hhtor-Index.db
+├── me-3gqe_1lnj_4sbpc2ezoscu9hhtor-Summary.db
+├── ...
+│
+├── ma-1abx_k29m_9fyug3sdtjwj8krpqh-big-Data.db
+├── ma-1abx_k29m_9fyug3sdtjwj8krpqh-Index.db
+├── ma-1abx_k29m_9fyug3sdtjwj8krpqh-Summary.db
+├── ...
+│
+└── ... (more SSTable components)
+```
+See the API [documentation](#copying-sstables-on-s3-backup) for more details about the actual backup request.
+
+3. `CREATE KEYSPACE` with S3 storage
+
+When creating a keyspace with S3 storage, the data is stored under the bucket passed as argument to the `CREATE KEYSPACE` statement.  
+Once the statement is issued, Scylla will use transparently the S3 bucket as the location of the SSTables for that keyspace.  
+Like in the case above, there is no hierarchy for the data, *all SSTables components are stored flat within the bucket*.
+```perl
+scylla-sstables-bucket/
+│
+├── me-3gqe_1lnj_4sbpc2ezoscu9hhtor-big-Data.db
```

Comment:
Is that really so? The `make_s3_object_name()` formats the name for keyspace storage options differently:

```c++
return format("/{}/{}/{}", _bucket, sst.generation(), sstable_version_constants::get_component_map(sst.get_version()).at(type));
```

so the name should rather be `scylla-sstables-bucket/3gqe_1lnj_4sbpc2ezoscu9hhtor/Data.db` or alike (I'm not sure about generation formatting

Reply:
you're very much correct, thanks! I just checked a S3 keyspace and indeed components are written according to `make_s3_object_name`, for some reason I thought I checked before, but it seems I verified for the backup case and not this one.